doc/releases/changelog-dev.md

```diff
@@ -78,6 +78,9 @@
 
 <h3>Internal changes ⚙️</h3>
 
+* `null.qubit` can now support an optional `track_resources` argument which allows it to record which gates are executed.
+  [(#1619)](https://github.com/PennyLaneAI/catalyst/pull/1619)
```

Comment:
I believe this is a new feature we should advertise :)

Reply:
Since this is an internal feature(at least for now), the approach we took for PL was not to openly advertise it. (See: https://github.com/PennyLaneAI/pennylane/pull/7226#discussion_r2068886788) Tagging @AntonNI8 for input

Reply:
Okay no worries, we can always update the changelog again on release. I know several users who'll want to use this though so I think we'll definitely want to advertise. tagging @josh146 @isaacdevlugt for the future